src/components/data-management/SpeciesSelector.jsx

```diff
@@ -3,20 +3,38 @@ import PropTypes from 'prop-types';
 import {
   Select, Typography, Space, Divider, Skeleton,
 } from 'antd';
-import _ from 'lodash';
 
 const { Text } = Typography;
 
 const SpeciesSelector = (props) => {
-  const { data, onChange } = props;
+  const { data, value, onChange } = props;
 
-  if (!data) {
+  if (!data || data.length === 0) {
     return <Skeleton.Input style={{ width: 300 }} size='small' />;
   }
 
+  const createLabel = (scientificName, displayName) => (
+    <Space direction='vertical'>
+      <Text type='primary'>{displayName}</Text>
+      <Text type='secondary'>
+        {scientificName}
+      </Text>
+    </Space>
+  );
+
+  const labeledValue = (organismId) => {
+    const organism = data.find((entry) => entry.id === organismId);
+
+    return {
+      value: organismId,
+      label: !organismId ? <></> : createLabel(organism.scientific_name, organism.display_name),
+    };
+  };
+
   return (
     <Select
-      onChange={(value, option) => onChange(option.displayName)}
+      value={labeledValue(value)}
```

Comment:
in here, if `value` is `null`, as it is set by default in default props, this line of code will fail

Reply:
No it won't. `organism` will be set to `undefined`, while `label` is set to an empty node. In the UI, this shows up as empty in the dropdown.

Reply:
I see. It seems to me a better practice to avoid using `null` as a default value, even if the code works with null, and the reason is that if we use null, then for situations when the value is null without us setting it to null by default, it will be harder to capture and debug. It also feels more messy. As a default value, I would use an empty string for example, or a string `"empty"` or something similar. What do you think?

Reply:
Using an empty string is cleaner. It's also easier to check with `!defaultValue`. I'll change the default value to ''.